version: '3.7'

volumes:
  grafana: {}

services:
  grafana:
    image: grafana/grafana:latest
    ports:
      - '3000:3000'
    volumes:
      - grafana:/var/lib/grafana
      - ./data:/app/data

  tor:
    image: getumbrel/tor:0.4.7.8@sha256:2ace83f22501f58857fa9b403009f595137fa2e7986c4fda79d82a8119072b6a
    user: "1000:1000"
    restart: on-failure
    volumes:
      - ./data/torrc:/etc/tor/torrc:ro
      - ./data/tor:/data
    environment:
      HOME: "/tmp"

  i2pd:
    image: purplei2p/i2pd:release-2.44.0@sha256:d154a599793c393cf9c91f8549ba7ece0bb40e5728e1813aa6dd4c210aa606f6
    user: "root"
    command: --sam.enabled=true --sam.address=0.0.0.0 --sam.port=7656 --socks.enabled=true --socks.address=0.0.0.0 --socks.port=4447 --loglevel=error
    restart: on-failure
    volumes:
      - ./data/i2pd:/home/i2pd/data
    ports:
      - 17656:7656
      - 14447:4447
